include/pybind11/cast.h

```diff
@@ -1338,14 +1304,15 @@ template <typename... Tuple> class type_caster<std::tuple<Tuple...>> {
         return true;
     }
 
-    static handle cast_impl(const type &, return_value_policy, handle,
+    template <typename T>
+    static handle cast_impl(T &&, return_value_policy, handle,
```

Comment:
Do you prefer keeping it in (and templated), or taking it out entirely, as in the last commit?